Test/Algorithms/Trainers/LogisticRegression.cpp

```diff
@@ -0,0 +1,157 @@
+	BOOST_CHECK_EQUAL(trainer.numberOfParameters(),2);
+	trainer.setLambda2(0.1);
+	BOOST_CHECK_EQUAL(trainer.lambda1(),0.0);
+	BOOST_CHECK_EQUAL(trainer.lambda2(),0.1);
+	BOOST_CHECK_EQUAL(trainer.parameterVector()(0),0.0);
+	BOOST_CHECK_EQUAL(trainer.parameterVector()(1),0.1);
+	BOOST_CHECK_EQUAL(trainer.accuracy(), 1.e-8);
+	LogisticRegression<>::ModelType model;
+	trainer.train(model,dataset);
+	
+	BOOST_CHECK_EQUAL(model.numberOfParameters(),11);
+	
+	//check gradient
+	{
+		CrossEntropy loss;
+		ErrorFunction error(dataset, &model.decisionFunction(),&loss);
+		RealVector derivative;
+		error.evalDerivative(model.parameterVector(),derivative);
+		RealVector penalty = 0.1 * model.parameterVector();
+		penalty(10) = 0;
+		BOOST_CHECK_SMALL(norm_inf(derivative+penalty),1.e-8);
+	}
+}
+
+BOOST_AUTO_TEST_CASE( LogReg_Binary_NoBias_NoLone){
+	PamiToy problem;
+	ClassificationDataset const& dataset = problem.generateDataset(50);
+	
+	LogisticRegression<> trainer(0,0.1,false);
+	BOOST_CHECK_EQUAL(trainer.lambda1(),0.0);
+	BOOST_CHECK_EQUAL(trainer.lambda2(),0.1);
+	BOOST_CHECK_EQUAL(trainer.accuracy(), 1.e-8);
+	LogisticRegression<>::ModelType model;
+	trainer.train(model,dataset);
+	
+	BOOST_CHECK_EQUAL(model.numberOfParameters(),10);
+	
+	//check gradient
+	{
+		CrossEntropy loss;
+		ErrorFunction error(dataset, &model.decisionFunction(),&loss);
+		RealVector derivative;
+		error.evalDerivative(model.parameterVector(),derivative);
+		RealVector penalty = 0.1 * model.parameterVector();
+		BOOST_CHECK_SMALL(norm_inf(derivative+penalty),1.e-8);
+	}
+}
+
+BOOST_AUTO_TEST_CASE( LogReg_Binary_Lone){
+	random::globalRng.seed(42);
+	PamiToy problem;
+	ClassificationDataset const& dataset = problem.generateDataset(50);
+	
+	double lambda1 = 0.1;
+	LogisticRegression<> trainer(lambda1,0.1);
+	BOOST_CHECK_EQUAL(trainer.lambda1(),lambda1);
+	BOOST_CHECK_EQUAL(trainer.lambda2(),0.1);
+	BOOST_CHECK_EQUAL(trainer.parameterVector()(0),lambda1);
+	BOOST_CHECK_EQUAL(trainer.parameterVector()(1),0.1);
+	BOOST_CHECK_EQUAL(trainer.accuracy(), 1.e-8);
+	LogisticRegression<>::ModelType model;
+	trainer.train(model,dataset);
+	
+	BOOST_CHECK_EQUAL(model.numberOfParameters(),11);
+	
+	//check gradient
+	{
+		RealVector point = model.parameterVector();
+		CrossEntropy loss;
+		ErrorFunction error(dataset, &model.decisionFunction(),&loss);
+		RealVector derivative;
+		error.evalDerivative(point,derivative);
+		RealVector penalty = 0.1 * point;
+		penalty(10) = 0;
+		derivative += penalty;
+		
+		//create subgradient derivative
+		for(std::size_t i = 0; i != 10; ++i){
+			if(std::abs(point(i)) > 1.e-13)
+				derivative(i) += lambda1 * (point(i) > 0? 1.0 : -1.0);
+			else//otherwise choose the proper subgradient that makes the resulting gradient smallest
+				derivative(i) -= std::min(lambda1,std::abs(derivative(i))) * (derivative(i) > 0? 1.0 : -1.0);
+		}
+		
+		std::cout<<point<<std::endl;
+		std::cout<<derivative<<std::endl;
```

Comment:
in case the test fails, i want to see that the last 5 features are reasonably close to 0. e.g. no bug but just wrong level of accuracy